spacy/pipeline/_parser_internals/_state.pxd

```diff
@@ -132,19 +135,24 @@ cdef cppclass StateC:
                 ids[i] = -1
 
     int S(int i) nogil const:
-        if i >= this._stack.size():
-            return -1
-        elif i < 0:
+        cdef int stack_size
+
+        stack_size = this._stack.size()
```

Comment:
Sorry, forgot this in the previous review, but I think if the assignment is also in the outer scope, it's nicer/shorter to assign at the declaration site.

```suggestion
        cdef int stack_size = this._stack.size()
```